packages/ember-views/lib/views/states/has_element.js

```diff
@@ -15,8 +16,21 @@ var hasElement = Object.create(_default);
 
 assign(hasElement, {
   $(view, sel) {
-    var elem = view.element;
-    return sel ? jQuery(sel, elem) : jQuery(elem);
+    if (isEnabled('ember-views-tagless-jquery') && view.tagName === '') {
+      var firstNode = view._renderNode.firstNode;
+      var lastNode = view._renderNode.lastNode;
+      var range = jQuery(firstNode).nextUntil(lastNode);
+
+      //Test if we should include first and last nodes in the range.
+      if (firstNode.nodeType !== 8) { range = range.add(firstNode); }
+      if (lastNode.nodeType !== 8) { range = range.add(lastNode); }
+
+      //Return inner find and topmost elements filter if a selector is provided
+      return sel ? jQuery(sel, range).add(range.filter(sel)) : range;
```

Comment:
It's a bit tricky. Example:
Suppose the sollowing HTML
```html
<ul>
  <li></li>
  <li class="second-li"></li>
  <li></li>
</ul>
```
Suppose `var lis` is the result of `$('li')`.
If you then to `$('.second-li', lis)`, the second li won't be returned. Basically, jquery "finds" if the inner elements match the selector, not if elements themselves match.

`.add(range.filter(sel))` will make sure we also consider the topmost elements that match `sel`.